### PR TITLE
fmt, vet and test code prior to building

### PR DESCRIPTION
**Pull request by mischief**

### CODE REVIEW

**Comment by pbx0** [comment]
lgtm

**Comment by mischief** [comment]
not done yet. i need to fix up the tests.

File: .travis.yml [hunk 1]
```diff
@@ -1,8 +1,16 @@
...
+  include:
+    - go: 1.4.2
+      install:
+        - go get golang.org/x/tools/cmd/cover
+        - go get golang.org/x/tools/cmd/vet
+    - go: 1.5.1
 
 # Gracefully handle developer forks
 install: |
+    go get golang.org/x/tools/cmd/vet
```
Comment (marineam) [inline .travis.yml]: why is this line here?

File: .travis.yml [hunk 2]
```diff
@@ -10,3 +18,8 @@ install: |
         export TRAVIS_BUILD_DIR="${MANTLE_DIR}"
         cd "${TRAVIS_BUILD_DIR}"
     fi
+
+script:
+  - ./test
```
Comment (marineam) [inline .travis.yml]: why this order?

File: .travis.yml (hunk 1 above)
Comment (mischief) [inline .travis.yml]: ptal 8-D

File: .travis.yml (hunk 2 above)
Comment (mischief) [inline .travis.yml]: no particular reason. should it be the other way?

Comment (marineam) [inline .travis.yml]: Not that it matters in this case because the two do not share artifacts but I just usually expect to see something built before tested :-P

**Comment by mischief** [comment]
@marineam ptal

File: build [hunk 3]
```diff
@@ -1,26 +1,16 @@
...
 
 if [[ $# -eq 0 ]]; then
 	set -- cmd/*
 fi
+
 for cmd in "$@"; do
 	cmd=$(basename "${cmd}")
 	echo "Building ${cmd}..."
-	go build -o "bin/${cmd}" "${REPO_PATH}/cmd/${cmd}"
+	go build -o "bin/${cmd}" -ldflags "${GLDFLAGS}" "${REPO_PATH}/cmd/${cmd}"
```
Comment (marineam) [inline build]: Why add ldflags?

File: env [hunk 4]
```diff
@@ -0,0 +1,18 @@
+#!/bin/bash -e
+
+ORG_PATH="github.com/coreos"
+PROJ="mantle"
+REPO_PATH="${ORG_PATH}/${PROJ}"
+VERSION=$(git describe --dirty --always)
+GLDFLAGS="-X github.com/coreos/fleet/version.Version=\"${VERSION}\""
```
Comment (marineam) [inline env]: fleet? delete please. If we want to start tagging and setting versions from tags that needs to be in a different PR

Comment (mischief) [inline env]: WOOPS :-)

**Comment by mischief** [comment]
@marineam ptal.

**Comment by marineam** [comment]
I still think that test script sucks but oh well.

LGTM

**Comment by mischief** [comment]
@marineam so does the whole godep/gopath symlink nonsense, but it's what we have. i just want to go fmt/vet/test in travis.

**Comment by marineam** [comment]
@mischief agreed, I don't use the build script either usually, it doesn't do incremental builds right so `go build` with a normal GOPATH setup is usually faster.
